packages/mobile/src/identity/verification.ts

```diff
@@ -534,52 +530,100 @@ function* isCodeAlreadyAccepted(code: string) {
   return existingCodes.find((c) => c.code === code)
 }
 
-function* revealNeededAttestations(
+function* revealAttestations(
   attestationsWrapper: AttestationsWrapper,
   account: string,
   phoneHashDetails: PhoneNumberHashDetails,
-  attestations: ActionableAttestation[],
-  withoutRevealing: boolean = false
+  attestations: ActionableAttestation[]
 ) {
-  Logger.debug(TAG + '@revealNeededAttestations', `Revealing ${attestations.length} attestations`)
-  const delayPeriod = 5000
+  Logger.debug(TAG + '@revealAttestations', `Revealing ${attestations.length} attestations`)
   let i = 0
+  const delayPeriod = 5000
+  for (const attestation of attestations) {
+    yield call(
+      revealAttestation,
+      attestationsWrapper,
+      account,
+      phoneHashDetails,
+      attestation,
+      // TODO (i1skn): remove this method and uncomment revealNeededAttestations above
+      // when https://github.com/celo-org/celo-labs/issues/578 is resolved
+      // send messages with 5000ms delay on Android
+      Platform.OS === 'android' ? delayPeriod * i++ : 0
+    )
+  }
+}
+function* completeAttestations(
+  attestationsWrapper: AttestationsWrapper,
+  account: string,
+  phoneHashDetails: PhoneNumberHashDetails,
+  attestations: ActionableAttestation[]
+) {
+  Logger.debug(
+    TAG + '@completeNeededAttestations',
+    `Completing ${attestations.length} attestations`
+  )
   yield all(
     attestations.map((attestation) => {
-      return call(
-        revealAndCompleteAttestation,
-        attestationsWrapper,
-        account,
-        phoneHashDetails,
-        attestation,
-        withoutRevealing,
-        // TODO (i1skn): remove this method and uncomment revealNeededAttestations above
-        // when https://github.com/celo-org/celo-labs/issues/578 is resolved
-        // send messages with 5000ms delay on Android
-        Platform.OS === 'android' ? delayPeriod * i++ : 0
-      )
+      return call(completeAttestation, attestationsWrapper, account, phoneHashDetails, attestation)
     })
   )
 }
 
-function* revealAndCompleteAttestation(
+export function* reRevealActionableAttestations() {
+  const contractKit = yield call(getContractKit)
+  const attestationsWrapper: AttestationsWrapper = yield call([
+    contractKit.contracts,
+    contractKit.contracts.getAttestations,
+  ])
+
+  const e164Number: string = yield select(e164NumberSelector)
+
+  const { phoneHashDetails }: VerificationState = yield select(verificationStateSelector)
+  const phoneHash = features.USE_PHONE_NUMBER_PRIVACY
+    ? phoneHashDetails.phoneHash
+    : getPhoneHash(e164Number)
+  const account: string = yield call(getConnectedUnlockedAccount)
+
+  const actionableAttestations: ActionableAttestation[] = yield call(
```

Comment:
So this appears to only try to re-reveal the already selected issuers (unless I'm missing something in the PR).

We also wanted to request an additional issuer for the case that the SMS could not be received. I think without a "successful response cache" or Tim's new `/get_attestations` endpoint yet, we could accomplish this by re-revealing to the original issuer, and if it returns an error then request a new issuer. WDYT?

Reply:
@aslawson  Yeap, you got it right! 

In this case it's pretty hard to predict if user has sufficient balance in advance. So, after user presses this button and we try to request another issuer with insufficient balance - we would show an error (or should we silently handle this and do nothing if balance is insufficient?). If we show an error, how can we explain to user why suddenly they need more funds?